Calendar/BookingHelpers/ExcelWrite.ps1

```diff
@@ -0,0 +1,123 @@
+
+    Write-Host "Exporting Services."
+    $Services = Get-GraphBookingsServices -Identity $Identity
+    $Services | Export-Excel -Path  $Path -WorksheetName "Services" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -Title "Services"
+
+    Write-Host "Exporting Custom Questions."
+    $CustomQuestions = Get-GraphBookingsCustomQuestions -Identity $Identity
+    $CustomQuestions | Export-Excel -Path  $Path -WorksheetName "Custom Questions" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -Title "Custom Questions"
+
+    Write-Host "Exporting MessageTrace for the past $($script:MessageTrackingDays) days."
+    $script:MessageTrackingLogs | Export-Excel -Path  $Path -WorksheetName "Message Traces" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3
+
+    Write-Host "Exporting Membership Log data to Excel."
+    $ArrayFromStaffLog = createArrayFromStaffLog -log $script:BookingStaffMembershipLog.MailboxLog
+    $ArrayFromStaffLog | Export-Excel -Path  $Path -WorksheetName "Membership Log" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize  -Title "Staff Membership Log"
+
+    Write-Host "Exporting Internal data to Excel."
+    $script:TenantSettings   | Export-Excel -Path  $Path -WorksheetName "Tenant Settings - Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-Title "Tenant Settings - Internal"
+    $script:BookingMBData  | Export-Excel -Path  $Path -WorksheetName "Mailbox Data - Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-Title "Mailbox Data - Internal"
+    $script:MBPermissions | Export-Excel -Path  $Path -WorksheetName "MB Permissions-Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-Title "MB Permissions - Internal"
+    $script:MBRecipientPermissions | Export-Excel -Path  $Path -WorksheetName "Recipient Permissions-Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-tableTitle "Recipient Permissions - Internal"
+
+    # $DateNow = Get-Date
+    # $DateStart = $DateNow.AddDays($script:MessageTrackingDays*-5)
+    # $Params = @{
+    #     staffIds      = @(
+    #         (Get-MgBookingBusinessStaffMember -BookingBusinessId $Identity).Id
+    #     )
+    #     startDateTime = @{
+    #         dateTime = $DateStart.ToString("yyyy-MM-ddT00:00:00")
+    #         timeZone = "UTC"
+    #     }
+    #     endDateTime   = @{
+    #         dateTime = $DateNow.ToString("yyyy-MM-ddT00:00:00")
+    #         timeZone = "UTC"
+    #     }
+    # }
+    # Get-MgBookingBusinessStaffAvailability -BookingBusinessId $Identity -BodyParameter $Params | Export-Excel -Path  $Path -WorksheetName "Availability 5 days" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd
```

Comment:
clean up comments

Reply:
done

Reply:
making comments unresolved till the commit is pushed.